ice_fishing_abm_1/ice_fishing_abm_gp/movement_destination_subroutine.py

```diff
@@ -1,12 +1,87 @@
+import numpy as np
+from sklearn.gaussian_process import GaussianProcessClassifier, GaussianProcessRegressor
+from sklearn.gaussian_process.kernels import RBF
+from .belief import generate_belief_mean_matrix
+class ExplorationStrategy:
+    def __init__(self, 
+        grid_size: int = 100,
+        ucb_beta=0.2, 
+        tau=0.01):
+    
+        self.grid_size = grid_size
+        self.ucb_beta = ucb_beta
+        self.softmax_tau = tau
+        self.mesh = np.array(np.meshgrid(range(self.grid_size), range(self.grid_size))).reshape(2, -1).T
+        self.mesh_indices = np.arange(0, self.mesh.shape[0])
+        self.belief_softmax = np.zeros((self.grid_size, self.grid_size))
+        self.other_agent_locs = np.empty((0, 2))
+        
+        
+    # Default algorithm selects destination randomly
+    def choose_destination(self):
+        self.destination = self.mesh[np.random.choice(self.mesh_indices, p=self.belief_softmax.reshape(-1)), :]
+    
+    def movement_destination(self):
```

Comment:
I think this is redundant we can just use choose destination method instead.

Reply:
accepted